graphql/resolvers.go

```diff
@@ -91,6 +91,26 @@ type queryResolver struct{ *Resolver }
 
 type patchResolver struct{ *Resolver }
 
+func (r *patchResolver) Duration(ctx context.Context, obj *restModel.APIPatch) (*PatchDuration, error) {
+	// excludes display tasks
+	tasks, err := task.Find(task.ByVersion(*obj.Id).WithFields(task.TimeTakenKey, task.StartTimeKey, task.FinishTimeKey))
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, err.Error())
+	}
+	if tasks == nil {
+		return nil, ResourceNotFound.Send(ctx, err.Error())
+	}
+	timeTaken, makespan := task.GetTimeSpent(tasks)
```

Comment:
the timeTaken and makespan fields are nullable, but I don't think this will ever be null. I think you'll need to add handling of zero time values returned by this function (you can use the util.IsZeroTime function), otherwise you won't be able to differentiate between a patch that didn't run and one whose makespan rounds to 0

Reply:
yeah good point. i was going to consider `0s` to be null since no task takes less than a second to run. this is a much better solution tho.